src/goto-symex/goto_symex_state.cpp

```diff
@@ -675,6 +675,8 @@ void goto_symex_statet::rename(
   if(!requires_renaming(type, ns))
     return; // no action
 
+  type = ns.follow(type);
+
```

Comment:
Is that not expanding some tag types?

Reply:
Intentionally so, with the idea to replace the expansion below (lines 728ff have been removed). But I might be missing something.

Reply:
Ok -- I'd move below the following `if` block, and then add a comment such as
`// expand struct and union tag types`